tests/unit/descriptors.cpp

```
@@ -4877,6 +4877,14 @@ TEST_F(NegativeDescriptors, DescriptorIndexingMissingFeatures) {
     TEST_DESCRIPTION("Use partially bound descriptor flag without feature.");
 
     AddRequiredExtensions(VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+    AddDisabledFeature(vkt::Feature::descriptorBindingPartiallyBound);
```

honestly for these "feature" specific tests, we could just do it the old fashion way, but happy for it to work either way

---

I think the root of the problem is that if you have a required feature, then all supported features are enabled in the same struct by the framework currently, which is unexpected.

If that will be fixed, then these `AddDisabledFeature` calls would not be necessary in any test case.

---

The problem is that previously the "old fashioned way" did not work together with the feature requirements approach, because the latter was simply ignored in that case, which does not interact well with "implicit feature requirements" corresponding to core versions.

We really need to get the two working together (by adding support for merging them), or abandon the old fashion way.

The previous behavior where the effects of `AddRequiredFeature` would simply be silently ignored if one specifies a create info pNext chain was simply broken (not all device-create structs are feature structs, and even if they are, it should be a test case error, not a silent ignorance of explicit feature requirements).